packages/@sanity/base/src/datastores/user/createUserStore.ts

```diff
@@ -1,122 +1,74 @@
-import {Observable, of, from} from 'rxjs'
+import {Observable, of, from, merge, defer, NEVER} from 'rxjs'
+import {catchError, map, mergeMap, mapTo, switchMap, shareReplay, tap, take} from 'rxjs/operators'
 import raf from 'raf'
 import DataLoader from 'dataloader'
-import pubsub from 'nano-pubsub'
 import authenticationFetcher from 'part:@sanity/base/authentication-fetcher'
-import {versionedClient} from '../../client/versionedClient'
-import {User, CurrentUser, CurrentUserEvent} from './types'
+import {observableCallback} from 'observable-callback'
+import {versionedClient as sanityClient} from '../../client/versionedClient'
+import {User, CurrentUser, UserStore, CurrentUserSnapshot} from './types'
 
-const userCache: Record<string, User | null> = {}
+const [logout$, logout] = observableCallback()
+const [refresh$, refresh] = observableCallback()
 
-const userChannel = pubsub<CurrentUser | null>()
-const errorChannel = pubsub<Error | null>()
-
-let _initialFetched = false
-let _currentUser: CurrentUser | null = null
-let _currentError: Error | null = null
-
-userChannel.subscribe((val: CurrentUser | null) => {
-  _currentUser = val
-
-  if (val) {
-    const normalized = normalizeOwnUser(val)
-    userCache.me = normalized
-    userCache[val.id] = normalized
+const userLoader = new DataLoader(
+  (userIds: readonly string[]) =>
+    fetchApiEndpoint<(User | null)[]>(`/users/${userIds.join(',')}`, {tag: 'users.get'})
+      .then(arrayify)
+      .then((response) => userIds.map((id) => response.find((user) => user?.id === id) || null)),
+  {
+    batchScheduleFn: (cb) => raf(cb),
```

Comment:
I'm guessing it's mostly to defer the loading to a reasonable timeframe, so `setTimeout` would probably work as well

Reply:
I'm all for changing to `setTimeout` (or just stick with whatever default dataloader uses) to avoid the external dependency, but let's stick with `raf` for now as it's been working so far :)